parlai/utils/curated_response.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import random
+
+
+"""
+Functions to generate safe curated response that repalce an Agent's main response.
+"""
+
+####################################################################################
+#
+#   Init prompt: initial message to start a chit-chat.
+
+STARTING_PROMPT_MESSAGES = (
+    "Hey I am here. Let's chat!",
+    "Hey, how are you doing today?",
+    "Hello, how is your day going so far?",
+    "Hi, I really like watchig movies, how about you?",
+    "Hi, are you into sports?",
+    "Have you seen any interesting movies recently?",
+)
+
+
+def generate_init_prompot():
+    return random.choice(STARTING_PROMPT_MESSAGES)
+
+
+####################################################################################
+#
+# Safe response
+#
+# A fall back method to avoid discussing sensetive and/or inappropriate topics.
+# This may be used to generate a safe response that substitues the main response from an agent,
+# in case the main response was flagged as inappropriate.
+
+# The generated responses consist of three parts.
+# 1-AVOID: Avoiding to continue the conversation about that topic, for example, "I don't know"
+# 2-TEMPLATE: a template to start talking about a new topic, for example, " have you heard about "
+# 3-TOPIC: the next topic to steer the conversation towards, for example, "new marvel movie"
+
+
+AVOID = (
+    "I don't know about that.",
+    "I am not sure about that.",
+    "I don't find that topic interesting.",
+    "Let's talk about something else.",
+)
+
+TEMPLATE = (
+    "Have you heard anything about",
+    "I was recently read about",
+    "I am really excited about",
+    "I know many people are intereseted in",
+    "I think it is excting to talk about",
+)
+
+TOPICS = (
+    "new marvel movies.",
+    "tonight game.",
+    "next year Oscar.",
+    "traveling to exotic destinations.",
+    "read new books.",
```

Comment:
"Have you heard anything about read new books."

Reply:
Yeah, it may generate non-sense like that too. I will have a second look and change them for a better template.